Exiled.Events/Patches/Events/Player/Transmitting.cs

```diff
@@ -60,20 +54,24 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
                 });
 
             // -2 to return false
-            newInstructions[newInstructions.Count - 2].WithLabels(retLabel);
+            newInstructions[newInstructions.Count - 1].WithLabels(retLabel);
 
             for (int z = 0; z < newInstructions.Count; z++)
                 yield return newInstructions[z];
 
             ListPool<CodeInstruction>.Shared.Return(newInstructions);
         }
 
-        private static bool HandleTransmitting(ReferenceHub hub, VoiceModuleBase voiceModule)
+        private static bool HandleTransmitting(PersonalRadioPlayback radioPlayback)
         {
-            if (hub == null || Player.Get(hub) is not Player player)
+            ReferenceHub hub = radioPlayback._owner;
+            if (hub == null || Player.Get(hub) is not Player player || Server.Host.ReferenceHub == hub)
+            {
                 return false;
+            }
 
-            TransmittingEventArgs ev = new(player, voiceModule);
+            IVoiceRole voiceModule = player.RoleManager.CurrentRole as IVoiceRole;
+            TransmittingEventArgs ev = new(player, voiceModule?.VoiceModule);
```

Comment:
you can remove `as` operator, and use IVoiceRole cast directly
```suggestion
            IVoiceRole voiceModule = (IVoiceRole)player.RoleManager.CurrentRole;
            TransmittingEventArgs ev = new(player, voiceModule.VoiceModule);
```